quil-py/src/instruction/classical.rs

```diff
@@ -0,0 +1,473 @@
+    #[new]
+    pub fn new(
+        py: Python<'_>,
+        memory_reference: PyMemoryReference,
+        operand: PyBinaryOperand,
+    ) -> PyResult<Self> {
+        Ok(Self((
+            MemoryReference::py_try_from(py, &memory_reference)?,
+            BinaryOperand::py_try_from(py, &operand)?,
+        )))
+    }
+
+    #[getter]
+    pub fn get_memory_reference(&self, py: Python<'_>) -> PyResult<PyMemoryReference> {
+        self.as_inner().0.to_python(py)
+    }
+
+    #[setter]
+    pub fn set_memory_reference(
+        &mut self,
+        py: Python<'_>,
+        memory_reference: PyMemoryReference,
+    ) -> PyResult<()> {
+        self.as_inner_mut().0 = MemoryReference::py_try_from(py, &memory_reference)?;
+        Ok(())
+    }
+
+    #[getter]
+    pub fn get_operand(&self, py: Python<'_>) -> PyResult<PyBinaryOperand> {
+        self.as_inner().1.to_python(py)
+    }
+
+    #[setter]
+    pub fn set_operand(&mut self, py: Python<'_>, binary_operand: PyBinaryOperand) -> PyResult<()> {
+        self.as_inner_mut().1 = BinaryOperand::py_try_from(py, &binary_operand)?;
+        Ok(())
+    }
+
+    pub fn __richcmp__(&self, py: Python<'_>, other: &Self, op: CompareOp) -> PyObject {
+        match op {
+            CompareOp::Eq => (self.as_inner() == other.as_inner()).into_py(py),
+            _ => py.NotImplemented(),
+        }
+    }
+}
+
+py_wrap_simple_enum! {
+    #[derive(Debug, PartialEq, Eq)]
+    PyBinaryOperator(BinaryOperator) as "BinaryOperator" {
+        And,
+        Ior,
+        Xor
+    }
+}
+impl_repr!(PyBinaryOperator);
+impl_str!(PyBinaryOperator);
+
+#[pymethods]
+impl PyBinaryOperator {
+    pub fn __richcmp__(&self, py: Python<'_>, other: &Self, op: CompareOp) -> PyObject {
+        match op {
+            CompareOp::Eq => (self.as_inner() == other.as_inner()).into_py(py),
+            _ => py.NotImplemented(),
+        }
+    }
+}
+
+py_wrap_data_struct! {
+    #[derive(Debug, PartialEq, Eq)]
+    #[pyo3(subclass)]
+    PyBinaryLogic(BinaryLogic) as "BinaryLogic" {
+        operator: BinaryOperator => PyBinaryOperator,
+        operands: BinaryOperands => PyBinaryOperands
+    }
+}
+impl_repr!(PyBinaryLogic);
+
+#[pymethods]
+impl PyBinaryLogic {
+    #[new]
+    pub fn new(
+        py: Python<'_>,
+        operator: PyBinaryOperator,
+        operands: PyBinaryOperands,
+    ) -> PyResult<Self> {
+        Ok(PyBinaryLogic(BinaryLogic::new(
+            BinaryOperator::py_try_from(py, &operator)?,
+            BinaryOperands::py_try_from(py, &operands)?,
+        )))
+    }
+
+    pub fn __richcmp__(&self, py: Python<'_>, other: &Self, op: CompareOp) -> PyObject {
+        match op {
+            CompareOp::Eq => (self.as_inner() == other.as_inner()).into_py(py),
+            _ => py.NotImplemented(),
+        }
+    }
+}
+
+py_wrap_data_struct! {
+    #[derive(Debug, PartialEq, Eq)]
+    #[pyo3(subclass)]
+    PyConvert(Convert) as "Convert" {
+        original: MemoryReference => PyMemoryReference,
```

Comment:
This used to be called `from`, but that didn't workout very well because `from` is a Python keyword. Open to other ideas for the naming here.

Reply:
In general, we probably don't want to to tailor the Rust library to suit Python, but now that I look at it, it seems like `CONVERT` could fix the same convention as other memory-related instructions like `MOVE`:

- `from` -> `source`
- `to` -> `destination`


This is of course a breaking change